tools/interactive/interactivetool_rstudio_bioc_3.20.xml

```diff
@@ -22,12 +22,29 @@
     </inputs>
     <command><![CDATA[
         #import re
-        echo "[`date`] - Setting up for RStudio." &&
+        echo "[`date`] - Setting up for RStudio as \$(id)." &&
+        ## Match the container user id to the host user/group id if supplied
```

Comment:
Can you do that on the TPV level (using --user)? If I remember correctly, none of our 40 ITs, including the normal RStudio IT, needed those hacks.

Reply:
The uid switching is being [done by TPV](https://github.com/galaxyproject/usegalaxy-playbook/blob/64616167a788a1170ca67d318a16be05be1c60dd/env/common/templates/galaxy/config/tpv/tools.yaml.j2#L230), and that's what caused a need for these changes.

My understanding why this works without `usermod` in the old wrapper is because of [777 permissions](https://github.com/hexylena/docker-rstudio-notebook/blob/14441bfbddf86969a209fdfa99185b25f0bf5f55/Dockerfile#L72) on the relevant folder, which I tried to avoid here.

Granted, these changes are needed only because of the .org infrastructure but the wrapper (and the image) work fine locally as well.

If there are other ideas for avoiding this, we can certainly explore that but I would like to avoid the 777 permissions approach.

Reply:
@bgruening given my comments, any thoughts? Is it acceptable to merge this?

Reply:
Please proceed however you want :) I'm not blocking this.

Reply:
Also, I prefer the user change here over world writable permissions on the output/working dirs. In addition to that generally being bad on a shared FS, it means I have to run cleanup scripts as root, or even worse hacks if the shared FS has root squashing (thankfully not the case in my scenario but it would be for others). This makes sure everything that escapes the container is still owned by the user running the container (unless they use sudo, I guess).